Tools/CISupport/ews-app/ews/views/results.py

```diff
@@ -50,19 +51,39 @@ def post(self, request):
             _log.error('Incorrect API Key {}. Host: {}. Ignoring data.'.format(data.get('EWS_API_KEY'), data.get('hostname')))
             return HttpResponse('Incorrect API Key received')
 
-        if data.get('type') == u'ews-build':
+        data.pop('EWS_API_KEY', None)
+
+        if not self.is_valid_result(data):
+            return HttpResponse('Incomplete data.')
+
+        if data.get('type') == 'ci-build':
+            return self.ci_build_event(data)
+
+        if data.get('type') == 'ews-build':
             return self.build_event(data)
 
-        if data.get('type') == u'ews-step':
+        if data.get('type') == 'ews-step':
             return self.step_event(data)
 
         _log.error('Unexpected data type received: {}'.format(data.get('type')))
         return HttpResponse('Unexpected data type received: {}'.format(data.get('type')))
 
-    def build_event(self, data):
-        if not self.is_valid_result(data):
-            return HttpResponse('Incomplete data.')
+    def ci_build_event(self, data):
+        _log.info(f'CI build event, data: {data}')
+
+        ci_system = data.get('ci_system', 'Misc')
+        commit_hash = data['commit_hash']
+        pr_number = data['pr_number']
+        pr_project = data.get('pr_project', 'WebKit/WebKit')
+        result = data['result']
+        url = data.get('url') or 'https://apple.com'
+
+        rc = CIBuild.save_build(commit_hash=commit_hash, build_id=data['build_id'], builder_display_name=data['builder_display_name'],
```

Comment:
nit: Can we retrieve the build_id and builder_display_name in the same way as the other data?